test/pipboy/parser/composite_readers_test.rb

```diff
@@ -7,14 +8,23 @@ def setup
   end
 
   def test_header
+    headerBytes = StringIO.new("\xA0\x00\x00\x00\x03")
+    assert_equal([160,3], Pipboy::Parser::CompositeReaders.header(headerBytes))
   end
 
   def test_list
+    primitiveReaders = Pipboy::Parser::PrimitiveReaders
+    streamBytes = StringIO.new("\x03\x00\x00\x00\x03\x48\x45\x4C\x4C\x4F\x57\x4F\x52\x4C")
+    assert_equal([1208156160, 1330400325, 1280462679], Pipboy::Parser::CompositeReaders.list(streamBytes, primitiveReaders, :uint32))
   end
 
   def test_dict_entry
+    streamBytes = StringIO.new("\xA0\x00\x00\x00\x03\x48\x45\x4C\x4C\x4F\x57\x4F\x52\x4C\x44\x70\x69\x70\x62\x6f\x79\x00\x00\x00\x00\x00")
+    assert_equal({"\u0003HELLOWORLDpipboy" => 160}, Pipboy::Parser::CompositeReaders.dict_entry(streamBytes))
   end
```

Comment:
Where's \u0003 coming from? That should be a clean string. Also not sure about the sample bytes. The string, for example, should be terminated with \x0 - check the primitive string reader